### ISSUE TITLE
Decktape does not render embedded images in SVGs

### ISSUE THREAD

**Issue by andrewrech** [astefanutti/decktape#37]
### Issue

Decktape does not load certain SVGs with reveal.js. Possibly related to #12?
### Minimal reproducible example

http://bit.ly/1QLgsv1
### My attempts to fix
- Linux vs. OSX
- Rebuilt phantom.js
- Reveal.js with no plugins / default themes
- Reveal.js local server vs. web 

Thank you in advance for considering this issue, and for this wonderful tool!
Andrew

**Comment by astefanutti** [comment]
Hi Andrew,

Thanks for the detailed report!

I've tried with PhantomJS `rastersize.js` example on the document below and the SVG is rendered correctly:
```
<html>
    <body>
        <div>
            <img src="test.svg">
        </div>
    </body>
</html>
```

So it looks like something happening with Reveal.js that prevents the SVG from loading / rendering correctly.

> Thank you in advance for considering this issue, and for this wonderful tool!

Thanks for the nice feedback. I'll dig into it ASAP. You may want to give a star, that helps prioritising things :wink:.

Antonin

**Comment by 6C3C41** [comment]
Great, let me know if there is anything I can do to help. Js experience minimal.

**Comment by astefanutti** [comment]
@6C3C41 no worries, I'll work on it ASAP.

**Comment by astefanutti** [comment]
@andrewrech, I get an error when I try to access the example that you provided. Would that be possible to re-upload it so that I can work on it?

**Comment by andrewrech** [comment]
@astefanutti Sure, here it is: https://s3.amazonaws.com/gobccuxmzacb4qcvlvisyq5vutramnv5ytshhilubpjpahiagt2ecjxvr1o9dnk/index.html#/

PDF output: https://db.tt/kF4z19yh

Thanks!

**Comment by astefanutti** [comment]
@andrewrech, thanks for your patience.

It happens that the issue originates from the SVG document included in slide 2 having an embedded image (base 64 encoded).

PhantomJS is not rendering the embedded image which is probably a bug. I've just created ariya/phantomjs#14348 to follow-up on this.

That may not be ideal, though as a work-around, you could directly include the embedded image in the slide rather than including the wrapping SVG, like:
```
<div class="reveal">
    <div class="slides">
        <section>
            <h1>SVG on following slide does not print</h1>
        </section>
        <section>
            <div class="stretch">
                <img alt="" src="data:image/png;base64,...==" height="100%">
            </div>
       </section>
    </div>
</div>
```
That works as expected.

An other alternative would be to use the `object` element to include the SVG document instead of the `img` element. Although the embedded image appears, it is rendered off the center. I suspect this is another issue PhantomJS faces with `viewBox` support.

**Comment by andrewrech** [comment]
Interesting - thank you so much for getting to the bottom of this!

My current workaround is to convert these offending images from R / ggplot2's pdf device to png using ImageMagick, which preserves transparency ect., if anyone has a similar issue.

`convert -trim -density 300 -flatten -background none input.pdf output.png`

**Comment by astefanutti** [comment]
Should be fixed in version `2.0.1` that uses headless Chrome. Let me know if you still face this issue.
